idp/idpclient/client.go

```diff
@@ -185,9 +189,7 @@ func (c *client) GetPrincipalById(ctx context.Context, systemBaseUri string, ten
 		return nil, fmt.Errorf("user is not allowed to invoke '%s'. Identityprovider returned HTTP-Statuscode '%d' and message '%s'",
 			resp.Request.URL, resp.StatusCode, responseMsg)
 	case http.StatusNotFound:
-		responseMsg, _ := ioutil.ReadAll(resp.Body)
```

Comment:
You must read the response body to EOF even if you don't need the reponse. cf. https://pkg.go.dev/net/http#Client
Add 
`_, _ = ioutil.ReadAll(resp.Body)`